pkg/cortex/cortex.go

```diff
@@ -229,18 +230,13 @@ func New(cfg Config) (*Cortex, error) {
 	}
 
 	cortex := &Cortex{
-		target: cfg.Target,
+		Cfg: cfg,
 	}
 
 	cortex.setupAuthMiddleware(&cfg)
 
-	serviceMap, err := cortex.initModuleServices(&cfg, cfg.Target)
-	if err != nil {
-		return nil, err
-	}
-
-	cortex.serviceMap = serviceMap
-	cortex.api.RegisterServiceMapHandler(http.HandlerFunc(cortex.servicesHandler))
+	mm := cortex.createModuleManager()
+	cortex.ModuleManager = mm
```

Comment:
It's very strange to call `createModuleManager` on cortex instance, only to put the result back into this instance.